src/driutils/benchmarking/partitioned_date_queries.py

```diff
@@ -0,0 +1,205 @@
+    # Fields of type DATE automatically picked up by duckdb so no need to specify as a hive type
+    return f"""
+        SELECT {COLUMNS_SQL}
+        FROM read_parquet('{base_path}/dataset={dataset}/*/*.parquet')
+        WHERE date BETWEEN '2019-01-01' AND '2019-12-31' AND SITE_ID='BUNNY'
+    """
+
+
+def query_multi_sites_and_multi_dates_using_hive_types_month(base_path, dataset):  # noqa: ANN001, ANN201
+    # Test larger and more complex query parameters
+    # Dates are hive types and filtered using BETWEEN
+    # Non partitioned column used
+    # Fields of type DATE automatically picked up by duckdb so no need to specify as a hive type
+    return f"""
+        SELECT {COLUMNS_SQL}
+        FROM read_parquet('{base_path}/dataset={dataset}/*/*.parquet')
+        WHERE date BETWEEN '2019-01-01' AND '2019-01-31'
+        AND SITE_ID IN ('BUNNY', 'ALIC1')
+    """
+
+
+def query_multi_sites_and_multi_dates_using_hive_types_year(base_path, dataset):  # noqa: ANN001, ANN201
+    # Test larger and more complex query parameters
+    # Dates are hive types and filtered using BETWEEN
+    # Non partitioned column used
+    # Fields of type DATE automatically picked up by duckdb so no need to specify as a hive type
+    return f"""
+        SELECT {COLUMNS_SQL}
+        FROM read_parquet('{base_path}/dataset={dataset}/*/*.parquet')
+        WHERE date BETWEEN '2019-01-01' AND '2019-12-31'
+        AND SITE_ID IN ('BUNNY', 'ALIC1')
+    """
+
+
+if __name__ == "__main__":
+    # Setup basic duckdb connection
+    conn = duckdb.connect()
+
+    conn.execute("""
+        INSTALL httpfs;
+        LOAD httpfs;
+        SET force_download = true;
+        SET enable_profiling = json;
+        SET profiling_output = 'profile.json';
+    """)
+
+    # Add s3 connection details
+    conn.execute("""
+        CREATE SECRET aws_secret (
+            TYPE S3,
+            PROVIDER CREDENTIAL_CHAIN,
+            CHAIN 'sts'
+        );
+    """)
+
+    queries = [
+        query_one_site_one_date(BASE_BUCKET_PATH, DATASET),
+        query_multi_dates_using_conditionals_month(BASE_BUCKET_PATH, DATASET),
+        query_multi_dates_using_conditionals_year(BASE_BUCKET_PATH, DATASET),
+        query_multi_sites_and_multi_dates_using_conditionals_month(BASE_BUCKET_PATH, DATASET),
+        query_multi_sites_and_multi_dates_using_conditionals_year(BASE_BUCKET_PATH, DATASET),
+        query_multi_dates_using_hive_types_month(BASE_BUCKET_PATH, DATASET),
+        query_multi_dates_using_hive_types_year(BASE_BUCKET_PATH, DATASET),
+        query_multi_sites_and_multi_dates_using_hive_types_month(BASE_BUCKET_PATH, DATASET),
+        query_multi_sites_and_multi_dates_using_hive_types_year(BASE_BUCKET_PATH, DATASET),
+    ]
+
+    # Create empty dataframe to store the results
+    data = pl.DataFrame()
+
+    for query in queries:
+        print(f"Running {query}\n")
+
+        # Query profile is saved to ./profile.json
+        conn.execute(query).pl()
```

Comment:
```suggestion
        conn.sql(query).pl()
```
probably not that important since we are migrating to the site partitioning anyway